airgun/widgets.py

```diff
@@ -1830,6 +1881,71 @@ def read(self):
         }
 
 
+class ProgressBarPF4(GenericLocatorWidget):
```

Comment:
here also if you can able to use same implementation from `ProgressBar` class inherit that